### PR TITLE
arraymath.jl autotangent 

### PR DESCRIPTION
**Pull request by mzgubic**
Helps with #360

### CODE REVIEW

File: test/rulesets/Base/mapreduce.jl [hunk 1]
```diff
@@ -2,39 +2,23 @@
     @testset "sum" begin
         sizes = (3, 4, 7)
         @testset "dims = $dims" for dims in (:, 1)
-            fkwargs = (dims=dims,)
             @testset "Array{$N, $T}" for N in eachindex(sizes), T in (Float64, ComplexF64)
                 s = sizes[1:N]
                 x = randn(T, s...)
```
Comment (oxinabox) [inline test/rulesets/Base/mapreduce.jl:7]:
```suggestion
                x = randn(T, sizes[1:N]...)
```

File: test/rulesets/Base/mapreduce.jl [hunk 2]
```diff
@@ -2,39 +2,23 @@
...
-                    # can't use frule_test here because it doesn't yet ignore nothing tangents
-                    y_ad, ẏ_ad = frule((Zero(), Zero(), ẋ), sum, abs2, x; dims=dims)
-                    @test y_ad == y
-                    ẏ_fd = jvp(_fdm, z -> sum(abs2, z; dims=dims), (x, ẋ))
-                    @test ẏ_ad ≈ ẏ_fd
-                end
-                @testset "rrule" begin
-                    rrule_test(sum, Δy, (abs2, nothing), (x, x̄); fkwargs=fkwargs)
-                end
+                x = randn(T, s...)
```
Comment (oxinabox) [inline test/rulesets/Base/mapreduce.jl:19]: `s` can be merged in

**Review by oxinabox — APPROVED** [review]
this is so much cleaner

**Comment by mzgubic** [comment]
Oh my bad I've started a branch off the previous one...This should only have arraymath.jl in it

**Review by oxinabox — APPROVED** [review]

**Pull request merged**